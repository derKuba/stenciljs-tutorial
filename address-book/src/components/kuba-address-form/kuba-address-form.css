:host {
  display: block;
  margin: 50px;
}

.kuba-input {
  margin-bottom: 25px;
  display: flex;
  justify-content: space-between;
  width: 400px;
}
