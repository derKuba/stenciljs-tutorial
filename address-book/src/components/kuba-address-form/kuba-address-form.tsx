import { Component, Host, h, State } from '@stencil/core';

@Component({
  tag: 'kuba-address-form',
  styleUrl: 'kuba-address-form.css',
  shadow: true,
})
export class KubaAddressForm {

  @State() innerText: string;

  onChange = ({ detail: { value } }: { detail: { value: string } }) => {
    console.log(value);
    this.innerText = value;
  }

  render() {
    return (
      <Host>
        <h2>Adress-Formular</h2>
        <kuba-input onInputEvent={this.onChange}></kuba-input>
        <kuba-button></kuba-button>
      </Host>
    );
  }

}
