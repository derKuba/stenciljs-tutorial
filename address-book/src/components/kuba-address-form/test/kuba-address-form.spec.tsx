import { newSpecPage } from "@stencil/core/testing";


jest.mock('../utils/create-uuid', () => {
  const originalModule = jest.requireActual('../utils/create-uuid');

  //Mock the default export and named export 'foo'
  return {
    __esModule: true,
    ...originalModule,

    create_UUID: () => "uuid",
  };
});

const addressStoreMock = {
  contacts: []
};
jest.mock("../../../store/address-store", () => ({
  __esModule: true,
  default: addressStoreMock
}));

import { KubaAddressForm } from "../kuba-address-form";


describe("kuba-address-form", () => {

  beforeEach(() => {
    addressStoreMock.contacts = []
  })

  it("renders", async () => {
    const page = await newSpecPage({
      components: [KubaAddressForm],
      html: `<kuba-address-form></kuba-address-form>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  // hier kommt man an die Grenze von der newSpecPage
  xit("should submit the form with query on dom", async () => {
    const page = await newSpecPage({
      components: [KubaAddressForm],
      html: `<kuba-address-form></kuba-address-form>`,
    });

    console.log(page.body.querySelector("kuba-address-form").shadowRoot.querySelector("kuba-button").innerHTML); // mÃ¶glich! output: speichern
    // page.body.querySelector("kuba-address-form").shadowRoot.querySelector("kuba-button").shadowRoot // undefined

    expect(true).toBe(true);
  });


  it("should submit the form", async () => {
    const addressForm = new KubaAddressForm();
    addressForm.firstNameState = "Max";
    addressForm.lastNameState = "Muster";
    addressForm.addressState = "HalloWeltWeg 23"
    addressForm.onSubmit();

    expect(addressStoreMock.contacts).toEqual([{ "address": "HalloWeltWeg 23", "firstname": "Max", "id": "uuid", "lastName": "Muster" }]);
  });

  it("should use a spy", async () => {
    const windowSpy = jest.spyOn(global.document, "getElementById");
    const addressForm = new KubaAddressForm();
    addressForm.onSubmit();

    expect(windowSpy).toBeCalled()
  });


});
