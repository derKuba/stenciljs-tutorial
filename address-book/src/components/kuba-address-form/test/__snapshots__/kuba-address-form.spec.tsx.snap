// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kuba-address-form renders 1`] = `
<kuba-address-form>
  <mock:shadow-root>
    <h1>
      Kontakt-Form
    </h1>
    <stencil-route-link activeclass="link-active" url="/">
      Zur√ºck
    </stencil-route-link>
    <hr>
    <kuba-input componentid="first-name" label="Vorname:"></kuba-input>
    <kuba-input componentid="last-name" label="Nachname:"></kuba-input>
    <div class="kuba-input">
      <label htmlfor="street">
        Adresse:
      </label>
      <input id="street" type="text">
    </div>
    <br>
    <kuba-button>
      speichern
    </kuba-button>
    <hr>
  </mock:shadow-root>
</kuba-address-form>
`;
